src/config.rs

```diff
@@ -53,28 +117,28 @@ impl Flow {
     }
 }
 
-#[derive(Default, Debug)]
+#[derive(Default, Debug, PartialEq)]
 pub struct ParallelConfig {
     pub recursive_steps: Flow,    // Multiple `StepCircuit`s.
     pub synthesis: Flow,          // Synthesis (within one `StepCircuit`)
     pub poseidon_witnesses: Flow, // The poseidon witness part of synthesis.
```

Comment:
Since we're here, can you turn those commentaries into proper docstrings for those struct fields please?